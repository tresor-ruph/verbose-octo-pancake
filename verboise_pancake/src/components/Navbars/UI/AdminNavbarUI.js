import React from 'react'
import {Dropdown } from "react-bootstrap";
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
// import Settings from 'components/Navbars/Settings'
import Settings from '../../Navbars/Settings'
const AdminNavbarUI = (props) => {
  const { handleLogOut, toggleOffcanvas, avatar, toggleClass, userName, showModal,onHide,handleSettings,profilePicture,handleRedirectHome } = props

  return (
    <div>
            <Settings show ={showModal} onHide={onHide} />

    <nav className="navbar col-lg-12 col-12 p-lg-0 fixed-top d-flex flex-row">
      <div className="navbar-menu-wrapper d-flex align-items-center justify-content-between ">
        <button className="navbar-toggler align-self-center " type="button" onClick={() => handleRedirectHome()}>
         verbosePancake
        </button>

        <ul className="navbar-nav navbar-nav-right">
          <li className="nav-item  nav-profile border-0">
            <Dropdown>
              <Dropdown.Toggle className="nav-link count-indicator bg-transparent">
                <img className="img-xs rounded-circle" src={ profilePicture|| avatar} alt="Profile" />
              </Dropdown.Toggle>
              <Dropdown.Menu className="preview-list navbar-dropdown pb-3">
                <Dropdown.Item className='admin-options user-option' >
                <FontAwesomeIcon  icon='user-alt'  size='sm'  style ={{marginRight: '10px'}}  /><span>{userName}</span>     
                </Dropdown.Item>
                <hr />
                <Dropdown.Item  className='admin-options' onClick={() => handleSettings()}>
                <FontAwesomeIcon  icon='cog'  size='sm'  style ={{marginRight: '10px'}}  /><span>Settings</span>     
                </Dropdown.Item>
                <hr />
                <Dropdown.Item  className='admin-options' onClick={() => handleLogOut()}>
                <FontAwesomeIcon  icon='sign-out-alt'  size='sm'  style ={{marginRight: '10px'}}  /><span>Sign out</span>     
                </Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
          </li>
        </ul>
     
      </div>
    </nav>
    </div>
  )

}

export default AdminNavbarUI