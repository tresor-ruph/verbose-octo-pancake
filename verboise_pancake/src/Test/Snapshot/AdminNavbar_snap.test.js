import React from 'react'
import renderer from 'react-test-renderer'
import AdminNavbarUI  from './../../components/Navbars/UI/AdminNavbarUI'

it('renders correctly', () => {
    const AdminNavbarSnap = renderer.create(<AdminNavbarUI />).toJSON();
    expect(AdminNavbarSnap).toMatchSnapshot();

});