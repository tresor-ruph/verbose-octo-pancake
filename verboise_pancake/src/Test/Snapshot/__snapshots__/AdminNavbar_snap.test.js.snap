// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <nav
    className="navbar col-lg-12 col-12 p-lg-0 fixed-top d-flex flex-row"
  >
    <div
      className="navbar-menu-wrapper d-flex align-items-center justify-content-between "
    >
      <button
        className="navbar-toggler align-self-center"
        onClick={[Function]}
        type="button"
      >
        <i
          className="mdi mdi-menu"
        />
      </button>
      <ul
        className="navbar-nav navbar-nav-right"
      >
        <li
          className="nav-item  nav-profile border-0"
        >
          <div
            className="dropdown"
          >
            <button
              aria-expanded={false}
              aria-haspopup={true}
              className="nav-link count-indicator bg-transparent dropdown-toggle btn btn-primary"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <img
                alt="Profile"
                className="img-xs rounded-circle"
              />
            </button>
          </div>
        </li>
      </ul>
      <button
        className="navbar-toggler navbar-toggler-right d-lg-none align-self-center"
        type="button"
      >
        <span
          className="mdi mdi-menu"
        />
      </button>
    </div>
  </nav>
</div>
`;
